x-pack/platform/plugins/shared/streams_app/public/types.ts

```diff
@@ -57,6 +58,7 @@ export interface StreamsAppStartDependencies {
   discoverShared: DiscoverSharedPublicStart;
   fieldFormats: FieldFormatsStart;
   fieldsMetadata: FieldsMetadataPublicStart;
+  indexLifecycleManagement: IndexLifecycleManagementPluginStart;
```

Comment:
is it required or optional? because in `x-pack/platform/plugins/shared/streams_app/kibana.jsonc` you declare it as optional. Here it's required. If it's optional it should be marked as optional here as well I think so that it can be properly guarded against. Because kibana will put `undefined` if it's missing and your types guarantee it here as present, which might blow up runtime